### PR TITLE
fix(core): reset table filters and sorters on current page re-navigate via side-nav

### LINKED ISSUE #6300: [BUG] not clear filters and sorters when clicking menu item
### Describe the bug

In a table, apply some filters then click the menu item in left side bar, the url should reset filters and sorters (only apply default ones)

### Steps To Reproduce

1.  go to https://example.admin.refine.dev/orders
2. apply status filter
3. click 'Orders' in left side bar

### Expected behavior

reset the filters

### Packages

```
    "@ant-design/charts": "^1.4.2",
    "@ant-design/graphs": "^1.4.0",
    "@ant-design/icons": "5.0.1",
    "@dnd-kit/core": "^6.1.0",
    "@dnd-kit/modifiers": "^7.0.0",
    "@dnd-kit/sortable": "^8.0.0",
    "@emotion/react": "^11.11.1",
    "@emotion/styled": "^11.11.0",
    "@googlemaps/react-wrapper": "^1.1.35",
    "@refinedev/antd": "^5.42.0",
    "@refinedev/cli": "^2.16.36",
    "@refinedev/core": "^4.53.0",
    "@refinedev/devtools": "^1.2.6",
    "@refinedev/graphql": "^6.5.4",
    "@refinedev/inferencer": "^4.6.6",
    "@refinedev/kbar": "1.3.12",
    "@refinedev/react-router-v6": "^4.5.11",
    "@refinedev/simple-rest": "^5.0.8",
    "@refinedev/supabase": "^5.9.2",
```

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by arndom**
- Update `useTable` hook to handle case where a user navigates to current page by clicking side-nav link intending to reset the filters and sorters.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://refine.dev/docs/guides-concepts/contributing/#commit-convention

## Bugs / Features

- [x] Related issue(s) linked
- [ ] Tests for the changes have been added
- [ ] Docs have been added / updated
- [x] Changesets have been added https://refine.dev/docs/guides-concepts/contributing/#creating-a-changeset

## What is the current behavior?
In a table, apply some filters/sorters then click the menu item in the sidebar, the URL should reset filters and sorters to the default but it doesn't.

This can be previewed at https://example.admin.refine.dev/orders

## What is the new behavior?
The above has been resolved. It resets the query params for filters and sorters back to their defaults.

fixes #6300

### CODE REVIEW

File: .changeset/spicy-flowers-allow.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+"@refinedev/core": major
```
Comment (BatuhanW) [inline .changeset/spicy-flowers-allow.md:2]:
```suggestion
"@refinedev/core": patch
```

**Review by BatuhanW — CHANGES_REQUESTED** [review]
Hey @arndom thanks for the PR, please make sure to add tests for the behavior.

**Review by aliemir — CHANGES_REQUESTED** [review]
Hey @arndom thank you for your effort! 🙏 I've left a comment in the issue (https://github.com/refinedev/refine/issues/6300#issuecomment-2406062566) about the implementation we're looking for. 

While your changes will work for the reported case, it won't cover other cases such as changing the query params without using the `useTable` hook. 

Let us know if you can work on the changes described in the comment 🙏

**Comment by arndom** [comment]
@aliemir Sure, it's best to finish what I started properly...though I may need some guidance as I'm still fresh on the codebase

**Comment by BatuhanW** [comment]
Hey @arndom are you gonna be able to continue on this PR? Please let us know if you need help with a specific issue. We'd be happy to guide you through.

**Comment by arndom** [comment]
@BatuhanW I still plan to, I just haven't had the time recently due to my work schedule. Some help on the approach to solving this would be great.
 
My current thought process: Since `syncWithLocation` defined in refine context is used by multiple implementations for params syncing, a change in all those may provide a fix (possibly a shared function). I am unsure about its efficiency or if there are better solutions as I haven't spent that much time in the repo.

**Comment by arndom** [comment]
@aliemir I believe the recent commits take into consideration the updating of filters and sorters; URL state -> internal state and vice versa.

**Comment by aliemir** [comment]
Hey @arndom, I've tried on my local but looks like the effect is exceeding the max depth and even if I ignore it, it starts looping when I have a change in filters 🤔 I'll try to look again to see what went wrong and at least provide tips to get it back on track 🙏

**Comment by arndom** [comment]
Sorry about that, seems I did not test properly, I'll go through it again and resolve the issues

**Comment by BatuhanW** [comment]
Hey @arndom feel free to re-open it when your changes are complete.

**Pull request closed without merging**